src/Symfony/Component/Translation/Dumper/XliffFileDumper.php

```diff
@@ -176,7 +176,7 @@ private function dumpXliff2(string $defaultLocale, MessageCatalogue $messages, ?
             $metadata = $messages->getMetadata($source, $domain);
 
             // Add notes section
-            if ($this->hasMetadataArrayInfo('notes', $metadata)) {
+            if ($this->hasMetadataArrayInfo('notes', $metadata) && !empty($metadata['notes'] ?? null)) {
```

Comment:
```suggestion
            if ($this->hasMetadataArrayInfo('notes', $metadata) && $metadata['notes'] !== []) {
```

We avoid `empty` usage, because its behavior too implicit.

Could you add a test case to cover this fix please?